### ISSUE TITLE
Improve accessible labels in CharacterInfoCard

### ISSUE THREAD

**Issue by vvornanen** [vvornanen/aoboshi#181]
See failing accessibility tests in Storybook. Tests of `CharactersCard` and `CharacterInfoCard` can be marked as ready after this is resolved.
